exporter/metric/internal/statsd/conn_test.go

```diff
@@ -0,0 +1,299 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statsd_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/otel/api/core"
+	"go.opentelemetry.io/otel/api/key"
+	"go.opentelemetry.io/otel/api/unit"
+	"go.opentelemetry.io/otel/exporter/metric/internal/statsd"
+	"go.opentelemetry.io/otel/exporter/metric/test"
+	export "go.opentelemetry.io/otel/sdk/export/metric"
+	"go.opentelemetry.io/otel/sdk/metric/aggregator/array"
+	"go.opentelemetry.io/otel/sdk/metric/aggregator/counter"
+	"go.opentelemetry.io/otel/sdk/metric/aggregator/gauge"
+)
+
+type testAdapter struct {
+	*statsd.LabelEncoder
+}
+
+func (*testAdapter) AppendName(rec export.Record, buf *bytes.Buffer) {
+	_, _ = buf.WriteString(rec.Descriptor().Name())
+}
+
+func (*testAdapter) AppendTags(rec export.Record, buf *bytes.Buffer) {
```

Comment:
question: Why do we need a different Adapter here?

Is there any chance that the test `Record` has labels encoded with another `LabelEncoder`? (since the only difference I could see between the real `Adapter` and the test `Adapter` is that one panics and another dont)

Reply:
Got it, looks better :)